Matrix API example crash

[MatrixApiActivity](https://github.com/mapbox/mapbox-android-demo/blob/0a39320f3acd450eb45d6cf34f479b775878974f/MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/javaservices/MatrixApiActivity.java) crashes on `master` when the activity tries to start up. It was working fine on [this branch/pr](https://github.com/mapbox/mapbox-android-demo/pull/643) but isn't working on `master`. Will look into it. Maybe it's a problem with how I'm casting?

cc @cammace  osana

```java
03-08 16:27:51.620 17083-17083/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                   Process: com.mapbox.mapboxandroiddemo, PID: 17083
                                                   java.lang.RuntimeException: Unable to start activity ComponentInfo{com.mapbox.mapboxandroiddemo/com.mapbox.mapboxandroiddemo.examples.javaservices.MatrixApiActivity}: com.google.gson.JsonParseException: java.lang.ClassNotFoundException: com.mapbox.geojson.Point
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2946)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3046)
                                                       at android.app.ActivityThread.-wrap11(Unknown Source:0)
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1688)
                                                       at android.os.Handler.dispatchMessage(Handler.java:105)
                                                       at android.os.Looper.loop(Looper.java:164)
                                                       at android.app.ActivityThread.main(ActivityThread.java:6809)
                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
                                                    Caused by: com.google.gson.JsonParseException: java.lang.ClassNotFoundException: com.mapbox.geojson.Point
                                                       at com.mapbox.d.b.e.a(Unknown Source:77)
                                                       at com.mapbox.d.b.e.deserialize(Unknown Source:0)
                                                       at com.google.gson.internal.bind.TreeTypeAdapter.read(Unknown Source:35)
                                                       at com.mapbox.d.l$a.a(Unknown Source:139)
                                                       at com.mapbox.d.l$a.read(Unknown Source:0)
                                                       at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(Unknown Source:2)
                                                       at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(Unknown Source:32)
                                                       at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(Unknown Source:0)
                                                       at com.mapbox.d.m$a.a(Unknown Source:119)
                                                       at com.mapbox.d.m$a.read(Unknown Source:0)
                                                       at com.google.gson.Gson.fromJson(Unknown Source:20)
                                                       at com.google.gson.Gson.fromJson(Unknown Source:4)
                                                       at com.google.gson.Gson.fromJson(Unknown Source:9)
                                                       at com.google.gson.Gson.fromJson(Unknown Source:0)
                                                       at com.mapbox.d.w.a(Unknown Source:48)
                                                       at com.mapbox.mapboxandroiddemo.examples.javaservices.MatrixApiActivity.l(Unknown Source:6)
                                                       at com.mapbox.mapboxandroiddemo.examples.javaservices.MatrixApiActivity.onCreate(Unknown Source:35)
                                                       at android.app.Activity.performCreate(Activity.java:6998)
                                                       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1230)
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2899)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3046) 
                                                       at android.app.ActivityThread.-wrap11(Unknown Source:0) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1688) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:105) 
                                                       at android.os.Looper.loop(Looper.java:164) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:6809) 
                                                       at java.lang.reflect.Method.invoke(Native Method) 
                                                       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767) 
                                                    Caused by: java.lang.ClassNotFoundException: com.mapbox.geojson.Point
                                                       at java.lang.Class.classForName(Native Method)
                                                       at java.lang.Class.forName(Class.java:453)
                                                       at java.lang.Class.forName(Class.java:378)
                                                       at com.mapbox.d.b.e.a(Unknown Source:39)
                                                       at com.mapbox.d.b.e.deserialize(Unknown Source:0) 
                                                       at com.google.gson.internal.bind.TreeTypeAdapter.read(Unknown Source:35) 
                                                       at com.mapbox.d.l$a.a(Unknown Source:139) 
                                                       at com.mapbox.d.l$a.read(Unknown Source:0) 
                                                       at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(Unknown Source:2) 
                                                       at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(Unknown Source:32) 
                                                       at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(Unknown Source:0) 
                                                       at com.mapbox.d.m$a.a(Unknown Source:119) 
                                                       at com.mapbox.d.m$a.read(Unknown Source:0) 
                                                       at com.google.gson.Gson.fromJson(Unknown Source:20) 
                                                       at com.google.gson.Gson.fromJson(Unknown Source:4) 
                                                       at com.google.gson.Gson.fromJson(Unknown Source:9) 
                                                       at com.google.gson.Gson.fromJson(Unknown Source:0) 
                                                       at com.mapbox.d.w.a(Unknown Source:48) 
                                                       at com.mapbox.mapboxandroiddemo.examples.javaservices.MatrixApiActivity.l(Unknown Source:6) 
                                                       at com.mapbox.mapboxandroiddemo.examples.javaservices.MatrixApiActivity.onCreate(Unknown Source:35) 
                                                       at android.app.Activity.performCreate(Activity.java:6998) 
                                                       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1230) 
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2899) 
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3046) 
                                                       at android.app.ActivityThread.-wrap11(Unknown Source:0) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1688) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:105) 
                                                       at android.os.Looper.loop(Looper.java:164) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:6809) 
                                                       at java.lang.reflect.Method.invoke(Native Method) 
                                                       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767) 
                                                    Caused by: java.lang.ClassNotFoundException: Didn't find class "com.mapbox.geojson.Point" on path: DexPathList[[zip file "/data/app/com.mapbox.mapboxandroiddemo-ov8q88fEVn6VCAUvZIfKXg==/base.apk"],nativeLibraryDirectories=[/data/app/com.mapbox.mapboxandroiddemo-ov8q88fEVn6VCAUvZIfKXg==/lib/arm64, /data/app/com.mapbox.mapboxandroiddemo-ov8q88fEVn6VCAUvZIfKXg==/base.apk!/lib/arm64-v8a, /system/lib64, /system/vendor/lib64]]
                                                       at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:93)
                                                       at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
                                                       at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
                                                       at java.lang.Class.classForName(Native Method) 
                                                       at java.lang.Class.forName(Class.java:453) 
                                                       at java.lang.Class.forName(Class.java:378) 
                                                       at com.mapbox.d.b.e.a(Unknown Source:39) 
                                                       at com.mapbox.d.b.e.deserialize(Unknown Source:0) 
                                                       at com.google.gson.internal.bind.TreeTypeAdapter.read(Unknown Source:35) 
                                                       at com.mapbox.d.l$a.a(Unknown Source:139) 
                                                       at com.mapbox.d.l$a.read(Unknown Source:0) 
                                                       at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(Unknown Source:2) 
                                                       at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(Unknown Source:32) 
                                                       at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(Unknown Source:0) 
                                                       at com.mapbox.d.m$a.a(Unknown Source:119) 
                                                       at com.mapbox.d.m$a.read(Unknown Source:0) 
                                                       at com.google.gson.Gson.fromJson(Unknown Source:20) 
                                                       at com.google.gson.Gson.fromJson(Unknown Source:4) 
                                                       at com.google.gson.Gson.fromJson(Unknown Source:9) 
                                                       at com.google.gson.Gson.fromJson(Unknown Source:0) 
                                                       at com.mapbox.d.w.a(Unknown Source:48) 
                                                       at com.mapbox.mapboxandroiddemo.examples.javaservices.MatrixApiActivity.l(Unknown Source:6) 
                                                       at com.mapbox.mapboxandroiddemo.examples.javaservices.MatrixApiActivity.onCreate(Unknown Source:35) 
                                                       at android.app.Activity.performCreate(Activity.java:6998) 
                                                       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1230) 
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2899) 
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3046) 
                                                       at android.app.ActivityThread.-wrap11(Unknown Source:0) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1688) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:105) 
                                                       at android.os.Looper.loop(Looper.java:164) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:6809) 
                                                       at java.lang.reflect.Method.invoke(Native Method) 
                                                       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767) 
```

--- a/MapboxAndroidDemo/proguard-rules.pro
+++ b/MapboxAndroidDemo/proguard-rules.pro
@@ -37,3 +37,6 @@
 
 -keep class com.segment.analytics.** { *; }
 -keep class com.mapbox.mapboxandroiddemo.model.usermodel.** { *; }
+
+# MAS data models that will be serialized/deserialized over Gson
+-keep class com.mapbox.services.api.directionsmatrix.v1.models.** { *; }

